switchapi/switch_pd_utils.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <openvswitch/vlog.h>
```

Comment:
(See previous comments regarding angle brackets vs. quotation marks)

The conventional order is:
1. The source file's own header file, if it has one. This helps verify that the header file defines all the symbols it uses, before it uses them.
2. System header files (in angle brackets)
3. Application header file (in quotation marks)

Each group of header files is frequently in lexicographical order. (Stratum does this.)

I like to insert a blank line between groups.

#ifdefed includes are a law unto themselves. Some people like to put them after the unconditional includes (they make the list harder to sort, and can make it harder to read). Other people like to preserve lexicographical order because it makes the files easier to diff. Both groups are right.

None of this is canonical unless the project's coding guidelines say otherwise.

The chief benefit is that it is easier to figure out if a header file has already been included when the #includes have a fixed order.